pkg/controller/node/node_controller.go

```diff
@@ -973,6 +988,47 @@ func (ctrl *Controller) updateCandidateMachines(pool *mcfgv1.MachineConfigPool,
 	return nil
 }
 
+// setUpdateInPrgressTaint applies in progress taint to all the nodes that are to be updated.
+// The taint on the individual node is removed by MCD once the update of the node is complete.
+// This is to ensure that the updated nodes are being preferred to non-updated nodes there by
+// reducing the number of reschedules.
+func (ctrl *Controller) setUpdateInProgressTaint(nodeName string) error {
+	return clientretry.RetryOnConflict(nodeUpdateBackoff, func() error {
+		oldNode, err := ctrl.kubeClient.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		oldData, err := json.Marshal(oldNode)
+		if err != nil {
+			return err
+		}
+
+		newNode := oldNode.DeepCopy()
+		if newNode.Spec.Taints == nil {
```

Comment:
should this be nil or len(newNode.Spec.Taints) == 0 

(goes back to my q about whether this will always be nil or if it can be empty)

Reply:
It depends on declaration. In this case, we have a slice but no backing array so nil is ok but there is nothing wrong in checking len as it works for both nil and empty slices. For example,

var a []string is a nil slice as it doesn't have underlying array
c := []string{} is an empty slice. 

Usually all the API objects returned will be nil slices as it encoding in JSON will be easier.